src/vehicle.cpp

```diff
@@ -521,7 +529,21 @@ void vehicle::init_state(int init_veh_fuel, int init_veh_status)
             }
         }
     }
-
+    // destroy tires until the vehicle is not drivable
+    if( destroyTires ) {
+        for( size_t p = 0; p < parts.size(); p++ ) {
```

Comment:
Doing it in a loop like that will make it always target some wheels first.
At this point all wheels should be in `wheelcache`. You could pick random wheels out of it with `random_entry` instead of iterating over all of them. This will return duplicates, but you can just treat them like the rest (setting part hp to 0 twice is no big deal).